spring-cloud-kubernetes-client-discovery/src/main/java/org/springframework/cloud/kubernetes/client/discovery/KubernetesInformerDiscoveryClient.java

```diff
@@ -89,13 +89,32 @@ public KubernetesInformerDiscoveryClient(SharedInformerFactory sharedInformerFac
 			Lister<V1Service> serviceLister, Lister<V1Endpoints> endpointsLister,
 			SharedInformer<V1Service> serviceInformer, SharedInformer<V1Endpoints> endpointsInformer,
 			KubernetesDiscoveryProperties properties) {
-		this.sharedInformerFactory = sharedInformerFactory;
-		this.serviceLister = serviceLister;
-		this.endpointsLister = endpointsLister;
+		this.sharedInformerFactories = List.of(sharedInformerFactory);
+		this.serviceListers = List.of(serviceLister);
+		this.endpointsListers = List.of(endpointsLister);
 		this.informersReadyFunc = () -> serviceInformer.hasSynced() && endpointsInformer.hasSynced();
 		this.properties = properties;
 	}
 
+	public KubernetesInformerDiscoveryClient(List<SharedInformerFactory> sharedInformerFactories,
```

Comment:
a new constructor that accepts List<Lister> now, instead of individual Lister. It actually accepts more items as lists here. We had a bug some time ago, where we found out that we need  `List<SharedInformer<V1Service>> serviceInformers` instead of a single `SharedInformer<V1Service>`. We learned about this the hard way, but at least now we know we need this.